src/vm/amd64/asmhelpers.S

```diff
@@ -140,30 +147,36 @@ NESTED_ENTRY ProfileLeaveNaked, _TEXT, NoHandler
-  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x10], rax    // caller rsp      -- struct probeRsp field
-  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x18], r10    // return address  -- struct ip field
-  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x20], rsi    //                 -- struct profiledRsp field
-  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x28], r11    // return value    -- struct rax field
-  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x30], r11    // r11 is null      -- struct hiddenArg field
-  movsd                   real8 ptr [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x38], xmm0    //      -- struct flt0 field
-  movsd                   real8 ptr [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x40], xmm1    //      -- struct flt1 field
-  movsd                   real8 ptr [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x48], xmm2    //      -- struct flt2 field
-  movsd                   real8 ptr [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x50], xmm3    //      -- struct flt3 field
+  mov                     [rsp +  0x0], r11    // r11 is null     -- struct functionId field
+  save_reg_postrsp        rbp, 0x8          //                 -- struct rbp field
+  mov                     [rsp + 0x10], rax    // caller rsp      -- struct probeRsp field
+  mov                     [rsp + 0x18], r10    // return address  -- struct ip field
+  mov                     [rsp + 0x20], rsi    //                 -- struct profiledRsp field
+  mov                     [rsp + 0x28], r11    // return value    -- struct rax field
+  mov                     [rsp + 0x30], r11    // r11 is null      -- struct hiddenArg field
+  movsd                   real8 ptr [rsp + 0x38], xmm0    //      -- struct flt0 field
+  movsd                   real8 ptr [rsp + 0x40], xmm1    //      -- struct flt1 field
+  movsd                   real8 ptr [rsp + 0x48], xmm2    //      -- struct flt2 field
+  movsd                   real8 ptr [rsp + 0x50], xmm3    //      -- struct flt3 field
+  movsd                   real8 ptr [rsp + 0x58], xmm4    //      -- struct flt4 field
+  movsd                   real8 ptr [rsp + 0x60], xmm5    //      -- struct flt5 field
+  movsd                   real8 ptr [rsp + 0x68], xmm6    //      -- struct flt6 field
+  movsd                   real8 ptr [rsp + 0x70], xmm7    //      -- struct flt7 field
   mov                     r10, 0x2  // PROFILE_LEAVE
-  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x58], r10d   // flags           -- struct flags field
+  mov                     [rsp + 0x78], r10d   // flags           -- struct flags field
 
   // we need to be able to restore the fp return register
-  save_xmm128_postrsp     xmm0, OFFSETOF_FP_ARG_SPILL +  0x8
+  save_xmm128_postrsp     xmm0, OFFSETOF_FP_ARG_SPILL_0
+  save_xmm128_postrsp     xmm0, OFFSETOF_FP_ARG_SPILL_1
```

Comment:
@janvorli many thanks for your comments. I've just committed the appropriate fixes